File: app/src/main/java/de/dreier/mytargets/base/adapters/dynamicitem/DynamicItemHolder.java
/*
 * Copyright (C) 2017 Florian Dreier
 *
 * This file is part of MyTargets.
 *
 * MyTargets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2
 * as published by the Free Software Foundation.
 *
 * MyTargets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package de.dreier.mytargets.base.adapters.dynamicitem;

import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public abstract class DynamicItemHolder<T> extends RecyclerView.ViewHolder {
    protected T item;

    public DynamicItemHolder(View itemView) {
        super(itemView);
    }

    public abstract void onBind(T item, int position, Fragment fragment, View.OnClickListener removeListener);
}

File: app/src/main/java/de/dreier/mytargets/base/adapters/dynamicitem/DynamicItemHolder.kt
/*
 * Copyright (C) 2018 Florian Dreier
 *
 * This file is part of MyTargets.
 *
 * MyTargets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2
 * as published by the Free Software Foundation.
 *
 * MyTargets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package de.dreier.mytargets.base.adapters.dynamicitem

import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import android.view.View

abstract class DynamicItemHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {
    protected var item: T? = null

    abstract fun onBind(
        item: T,
        position: Int,
        fragment: Fragment,
        removeListener: View.OnClickListener
    )
}
